src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -411,16 +409,32 @@ private DeclarationModifiers MakeModifiers(TypeKind typeKind, DiagnosticBag diag
                         var mods = singleDeclaration.Modifiers;
                         if ((mods & DeclarationModifiers.Partial) == 0)
                         {
-                            diagnostics.Add(ErrorCode.ERR_MissingPartial, singleDeclaration.NameLocation, self.Name);
+                            diagnostics.Add(ErrorCode.ERR_MissingPartial, singleDeclaration.NameLocation, this.Name);
                             modifierErrors = true;
                         }
                     }
                 }
             }
 
+            if (this.Name == SyntaxFacts.GetText(SyntaxKind.RecordKeyword))
+            {
+                var location = declaration.Declarations[0].NameLocation;
```

Comment:
Would this not issue an error if you had a partial declaration where the "first" declaration was escaped, but the second one wasn't? Consider testing something like:
```cs
partial class @record {}
partial class record {}
```

Reply:
Actually, I'm not even sure that this code is necessary? It seems that the only direct implementations of this type are `SourceNamedTypeSymbol` (already modified in this PR), `ImplicitNamedTypeSymbol` (which does not have a user-controlled name), and `SimpleProgramNamedTypeSymbol` (also doesn't have a user-controlled name). Do we have a test case that fails if this is removed?

---
In reply to: [477450215](https://github.com/dotnet/roslyn/pull/47094#discussion_r477450215) [](ancestors = 477450215)

Reply:
Many tests fail without this logic. This is responsible for checking the name of the type. The logic added in `SourceNamedTypeSymbol` is responsible for checking type parameters.

For instance this test fails:
```
        public void TypeNamedRecord()
        {
            var src = @"
class record { }

class C
{
    record M(record r) => r;
}";
            var comp = CreateCompilation(src);
...
```

---
In reply to: [477453969](https://github.com/dotnet/roslyn/pull/47094#discussion_r477453969) [](ancestors = 477453969,477450215)